### PR TITLE
Allow iframe links to open on android through a parameter

### PR DESCRIPTION
**Pull request by cnico**
## Breaking change

None

## Proposed change

Following the discussion in https://github.com/home-assistant/android/issues/3044 and my previous PR in https://github.com/home-assistant/frontend/pull/14379, I propose the approach of adding the 'allow-top-navigation-by-user-activation' value for the sandbox's iframe attribute.
By default, it is not allowed and if the user chooses it, it is allowed.

So it resolves my initial problem of not being able to clic on the links of the iframe in my android device. 
I tested it in developpement mode and it works well.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Create an iframe card in lovelace with the following src url : https://www.programme-television.org/?no_layout=
Check or un check the new option and see in debug mode, the presence or absence of the sandbox="allow-top-navigation-by-user-activation" value.
Open it on your android device and clic on a link of the tv show : it opens correctly now in a tab of the default android browser.

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/android/issues/3044
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by alanmilinovic — DISMISSED** [review]
I tested the change and it is working fine.

**Comment by alanmilinovic** [comment]
Wow how much time is needed for a simple code review.

**Comment by bramkragten** [comment]
> Wow how much time is needed for a simple code review.

It is not perse the code review, more if we want to add this feature...

File: src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts [hunk 1]
```diff
@@ -26,6 +27,7 @@ const SCHEMA = [
     schema: [
       { name: "url", required: true, selector: { text: {} } },
       { name: "aspect_ratio", selector: { text: {} } },
+      { name: "allow_open_top_navigation", selector: { boolean: {} } },
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts:30]: We should not add this option to the UI

File: src/translations/en.json [hunk 2]
```diff
@@ -4234,6 +4234,7 @@
             },
             "generic": {
               "alt_text": "Alternative Text",
+              "allow_open_top_navigation": "Allow open iframe links",
```
Comment (bramkragten) [inline src/translations/en.json:4237]:
```suggestion
```

File: src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts (hunk 1 above)
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts:30]:
```suggestion
```

**Comment by bramkragten** [comment]
Please add a documentation PR for this option

**Comment by cnico** [comment]
@bramkragten , I did not know we could set options via the code editor but not via the UI. 
I removed the UI config options as you suggest.
I also synced my fork to the upstream and tested it. It works via configuration in the code editor.

**Comment by cnico** [comment]
And finaly added the documentation PR also.

File: src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts (hunk 1 above)
Comment (cnico) [reply src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts]: Done

File: src/translations/en.json (hunk 2 above)
Comment (cnico) [reply src/translations/en.json]: Done

File: src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts (hunk 1 above)
Comment (cnico) [reply src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts]: Done

File: src/panels/lovelace/cards/hui-iframe-card.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export class HuiIframeCard extends LitElement implements LovelaceCard {
       type: "iframe",
       url: "https://www.home-assistant.io",
       aspect_ratio: "50%",
+      allow_open_top_navigation: false,
```
[the remark this answers is not in the document]
Comment (cnico) [reply src/panels/lovelace/cards/hui-iframe-card.ts]: Done

**Review by bramkragten — APPROVED** [review]

**Comment by alanmilinovic** [comment]
Will this change be in a next release? Thank you.

**Pull request merged**